qiskit_experiments/library/randomized_benchmarking/layer_fidelity.py

```diff
@@ -454,16 +506,21 @@ def __circuit_body(
                 sample = rng.integers(NUM_1Q_CLIFFORD)
                 cliffs_1q[k] = compose_1q(cliffs_1q[k], sample)
                 circ._append(_to_gate_1q(sample), (circ.qubits[q],), ())
-            circ._append(barrier_inst)
+            for barrier_inst in barrier_inst_lst:
+                circ._append(barrier_inst)
             # add two qubit gates
             for j, qpair in enumerate(two_qubit_layer):
                 circ._append(gate2q, tuple(circ.qubits[q] for q in qpair), ())
                 cliffs_2q[j] = compose_2q(cliffs_2q[j], gate2q_cliff)
                 # TODO: add dd if necessary
             for k, q in enumerate(one_qubits):
                 # TODO: add dd if necessary
-                pass
-            circ._append(barrier_inst)
+                # if there is a min_delay, just need
+                # to add to one of the qubits
+                if min_delay is not None and k == 0:
+                    circ.delay(min_delay, q)
```

Comment:
It would be  easy to warn `if min_delay is not None and not one_qubits` at the top of the function if we wanted.

Also, this is more work but a different way to do this feature would be to take a qubit pair as `replace_with_delays` instead of `min_delay` and have the experiment look up `self.backend.target[opts.two_qubit_gate][opts.replace_with_delays].duration` and insert a delay with that duration instead of the two qubit gate for that pair. Maybe you need one pair per layer. That doesn't give the flexibility to vary the duration though, but it does model the desired behavior of replacing a gate with a delay more closely.

Reply:
I added some warnings. Interesting idea, but for flexibility I'll just leave it as is.